src/quote/swap/adapters/static/transaction.ts

```diff
@@ -49,7 +49,7 @@ export function buildTransaction(
           indexToken.address,
           outputAmount,
           inputToken.address,
-          quoteAmount,
+          inputAmount,
```

Comment:
main change: use input token amount instead of quote amount (for leveraged tokens) when building tx object